File: 백준/Bronze/2884. 알람 시계/알람 시계.cc
#include <stdio.h>

int main(void) {
    int h, m;
    scanf("%d %d", &h, &m);

    int result = h * 60 + m - 45;
    if (result < 0) result += 1440;

    printf("%d %d", result / 60, result % 60);
    return 0;
}

File: 백준/Bronze/2884. 알람 시계/알람 시계.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int h = sc.nextInt();
        int m = sc.nextInt();

        int result = h * 60 + m - 45;
        if (result < 0) result += 1440;

        System.out.println((result / 60) + " " + (result % 60));
    }
}